### ISSUE TITLE
Contributor/Supporter - Account - Connect Stellar account to receive lumens or incentivize projects

### ISSUE THREAD

**Issue by Mcscott142** [revelrylabs/incentivize#55]
## Background 
Each contributor needs a Stellar Account in order to send and receive payments. Upon signing in with GitHub an account needs to be created or connected for the user. 

**Note** Anytime someone wants to deposit lumens they need a public and private key. 

#### Relevant Links
https://www.stellar.org/developers/guides/get-started/create-account.html
https://www.stellar.org/developers/guides/get-started/transactions.html

## Scenario: As a Contributor, I connect my Stellar Account on my Account Page
- [ ] When I select connect Stellar Account
- [ ] Then I am able to put in my existing public key
- [ ] And I am informed that my Stellar Account is connected

**Comment by bryanjos** [comment]
@Mcscott142 should this mean user should be sent to account page after logging in for now?

**Comment by thomasknoll** [comment]
assuming we could ship simply a field for inputting a lumen wallet public address - and then enhance with the ability to create accounts, wallets, and other fanciness.

**Comment by Mcscott142** [comment]
@bryanjos I think that makes the most sense to send them to the account page. 

@thomasknoll That is true, if creating a Stellar account is too big of a lift we could just have the connect part to begin with.

**Comment by thomasknoll** [comment]
to be fair, I assume we will be able to get to it! Just saying we should be able to ship with a field for wallet address

**Comment by bryanjos** [comment]
A little bit of a snag:
https://www.stellar.org/developers/guides/get-started/create-account.html

```
Now that you have a seed and public key, you can create an account. In order to prevent people from making a huge number of unnecessary accounts, each account must have a minimum balance of 1 lumen (lumens are the built-in currency of the Stellar network).[2] Since you don’t yet have any lumens, though, you can’t pay for an account. In the real world, you’ll usually pay an exchange that sells lumens in order to create a new account.[3] On Stellar’s test network, however, you can ask Friendbot, our friendly robot with a very fat wallet, to create an account for you.

To create a test account, send Friendbot the public key you created. It’ll create and fund a new account using that public key as the account ID.
```

**Comment by bryanjos** [comment]
So we do have to supply each account with at least a lumen

**Comment by thomasknoll** [comment]
is that a 'heads up' or a 'blocker' ?

**Comment by bryanjos** [comment]
It means that the account isn't technically generated and in the Stellar Network. Will just be randomly generated keys until the first transaction

**Comment by thomasknoll** [comment]
1. seems fine, or maybe I'm just not understanding the scope of this?
2. if we need to, we can definitely throw a lumen or two in accounts?

**Comment by bryanjos** [comment]
Everything described in this issue works otherwise. So it can be a new ticket and a different concern

**Comment by thomasknoll** [comment]
hotness

**Comment by Mcscott142** [comment]
#85 to continue discussion started on account generation/activation

**Comment by brittanygay** [comment]
Jumping on @bryanjos's ticket to make a few design updates

**Comment by brittanygay** [comment]
Styles have been updated. This is now approved by me.

**Comment by Mcscott142** [comment]
Account info being cut off by NavBar

![incentivize](https://user-images.githubusercontent.com/6877299/43806401-67439536-9a69-11e8-81a4-0b99ffd9d3e3.png)

**Comment by brittanygay** [comment]
@Mcscott142 I actually just put in a quick fix for this but it hasn't gotten merged yet. Once my edit is merged and reflected on staging I'll pass back to you.

**Comment by brittanygay** [comment]
Style fix is reflected on staging, this is now ready for QA.

**Comment by Mcscott142** [comment]
Removing the second set of AC. Making an enhancement ticket with it. 
```
## Scenario: As a Contributor, I create a Stellar Account on my Account Page
- [ ] When I select to create a Stellar account 
- [ ] Then my stellar account is created
- [ ] And I see my public and private key with a warning message: 
> A Stellar Account comes with 2 keys. A public key that serves as a way for you to receive Lumens (Think Venmo username or email for PayPal) and Secret Key that gives you access to your account. This is the one and only time we will show you the secret key, please save it somewhere safe as anyone who has access to it has complete access to your account.
```

**Comment by Mcscott142** [comment]
So I have to have Lumen's in my Stellar account for this to connect right? @bryanjos

**Comment by bryanjos** [comment]
yes if I understand what you are asking.

**Comment by bryanjos** [comment]
But also no

**Comment by Mcscott142** [comment]
Steps I am taking
1) I go to https://www.stellar.org/account-viewer/#!/. 
2) I get my public/secret keys
3) I use my GitHub email and public stellar key on https://incentivize.stage.revelry.net/account/edit
4) Nothing happens

Is that happening because I don't have Lumens?

**Comment by bryanjos** [comment]
It has nothing to do with having lumens or not. Nothing happens at that point. If the info saved, it should be good to go

**Comment by Mcscott142** [comment]
Looks like a JS console error on submit. Moving back. 

![fullscreen_8_8_18__4_38_pm](https://user-images.githubusercontent.com/6877299/43865880-7b584286-9b29-11e8-871e-73666f1730fd.png)

**Comment by bryanjos** [comment]
That's not a JavaScript error. That's a unauthorized error with typekit. Should not effect this page at all but something to tell Brittany about

**Comment by Mcscott142** [comment]
@bryanjos  So I'm confused at what's wrong here or if anything is wrong. Is the typekit problem stopping me from connecting my account? When I hit submit with an email and my public key what currently happens? 

What I'm expecting is some sort of message that tells me my account is connected and then I'm still on my account page.

**Comment by bryanjos** [comment]
I will add a message

**Comment by gerardramos** [comment]
will create an enhancement ticket to update and have a better flash message.
